mgmtfn/k8splugin/contivk8s/clients/network.go

```diff
@@ -0,0 +1,122 @@
+/***
+Copyright 2015 Cisco Systems Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clients
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	log "github.com/Sirupsen/logrus"
+	"github.com/contiv/netplugin/mgmtfn/k8splugin/cniapi"
+	"io/ioutil"
+	"net"
+	"net/http"
+)
+
+const (
+	nwURL = "http://localhost"
+)
+
+// NWClient defines informatio needed for the k8s api client
+type NWClient struct {
+	baseURL string
+	client  *http.Client
+}
+
+func unixDial(proto, addr string) (conn net.Conn, err error) {
+	sock := cniapi.ContivCniSocket
+	return net.Dial("unix", sock)
+}
+
+// NewNWClient creates an instance of the network driver client
+func NewNWClient() *NWClient {
+	c := NWClient{}
+	c.baseURL = nwURL
+
+	transport := &http.Transport{Dial: unixDial}
+	c.client = &http.Client{Transport: transport}
+
+	return &c
+}
+
+// AddPod adds a pod to contiv using the cni api
+func (c *NWClient) AddPod(podInfo interface{}) (*cniapi.RspAddPod, error) {
```

Comment:
Because we moved the kubeserver access inside netplugin, these functions will need access to a kube server. I think sys-test is the only option. Will add it after vagrant work is completed.